.github/workflows/mac.yml

```diff
@@ -0,0 +1,122 @@
+name: MACOS
```

Comment:
In general, I wonder if we can nudge this to be even more similar to the linux action, just for long term maintenance purposes

Reply:
Seconded. I wonder if a  job parameter indicating the OS type and some [conditionals](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif) in the job steps that check that parameter 
would do the trick.